### PR TITLE
Process patient for COPD codings, and add marker if found

### PR DESCRIPTION
**Pull request by pbugni**

### CODE REVIEW

File: carl/modules/codeableconcept.py [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+        codeable concept.  Therefore, an exception is raised if the
+        instance doesn't contain exactly one coding.
+
+        """
+        if len(self.codes) != 1:
+            raise ValueError(
+                f"Require single coding to include {self} in search for value_param ")
+
+        coding = self.codes[0]
+        return '|'.join((coding.system, coding.code))
```
Comment (ivan-c) [inline carl/modules/codeableconcept.py]: now I can nitpick about f-strings ;)
```suggestion
        return f"{coding.system}|{coding.code}")
```

File: carl/views.py [hunk 2]
```diff
@@ -75,3 +77,24 @@ def default(self, obj):
 
     return jsonify(settings)
 
+
+@base_blueprint.route('/process/<int:patient_id>')
```
Comment (ivan-c) [inline carl/views.py]: nice! helpful debugging hook

File: tests/test_modules/valueset_bundle.json [hunk 3]
```diff
@@ -0,0 +1,155 @@
...
+  "resourceType": "Bundle",
+  "id": "81e8eeb8-573b-4b6c-b3e3-1cc6e6dc9e48",
+  "meta": {
+    "lastUpdated": "2021-11-04T23:28:32.722+00:00"
+  },
+  "type": "searchset",
+  "total": 1,
+  "link": [ {
+    "relation": "self",
+    "url": "http://fhir.pb.dev.cnics.cirg.washington.edu/fhir/ValueSet?url=http%3A%2F%2Fcnics-cirg.washington.edu%2Ffhir%2FValueSet%2FCNICS-COPD-codings"
```
Comment (ivan-c) [inline tests/test_modules/valueset_bundle.json:11]:
```suggestion
    "url": "http://fhir.pb.dev.cnics.cirg.washington.edu/fhir/ValueSet?url=http://cnics-cirg.washington.edu/fhir/ValueSet/CNICS-COPD-codings"
```

File: carl/views.py [hunk 4]
```diff
@@ -75,3 +77,24 @@ def default(self, obj):
...
+        "patient_id": patient_id,
+        "COPD codings found": len(positive_codings) > 0}
+    if positive_codings:
+        condition = Condition()
+        condition.code = CodeableConcept(CNICS_COPD_coding)
+        condition.subject = Patient(patient_id)
+        response = persist_resource(resource=condition)
+        results['condition'] = response
+
+    return results
```
Comment (ivan-c) [inline carl/views.py]: minor nitpick; flipped the check to exit early
```suggestion
    if not positive_codings:
        return results

    condition = Condition()
    condition.code = CodeableConcept(CNICS_COPD_coding)
    condition.subject = Patient(patient_id)
    response = persist_resource(resource=condition)
    results['condition'] = response

    return results
```

**Review by ivan-c — APPROVED** [review]
looks good a first glance- trying to wrap my head around the workflow

File: carl/views.py [hunk 5]
```diff
@@ -27,6 +27,7 @@ def bootstrap():
             except json.decoder.JSONDecodeError as je:
```
Comment (ivan-c) [inline carl/views.py]: Might be nice to wrap this into a function, for importing into views, and a flask CLI

File: carl/modules/codeableconcept.py (hunk 1 above)
Comment (pbugni) [reply carl/modules/codeableconcept.py]: GREAT - that's my pref, put that `join` in there for you :)

File: tests/test_modules/valueset_bundle.json (hunk 3 above)
Comment (pbugni) [reply tests/test_modules/valueset_bundle.json:11]: Those file contents come directly from a HAPI query, and IMO, match the HTTP spec, needing to URL encode query string parameters.  You sure you want that change?

File: carl/views.py (hunk 5 above)
Comment (pbugni) [reply carl/views.py]: Done - better?

File: tests/test_modules/valueset_bundle.json (hunk 3 above)
Comment (ivan-c) [reply tests/test_modules/valueset_bundle.json:11]: Good points, thanks!

**Review by ivan-c — APPROVED** [review]

**Pull request merged**